state-chain/pallets/cf-swapping/src/lib.rs

```diff
@@ -1964,6 +1983,18 @@ pub mod pallet {
 				swap_amount
 			};
 
+			// Restrict the number of chunks based on the minimum chunk size.
+			let dca_params = dca_params.map(|mut dca_params| {
```

Comment:
@zoheb391 just double checking that the frontend will be able to handle this (changing the number of dca chunks based on the amount)? I imagine if you are reading the number of chunks from `SwapRequested` rather than by looking at the channel, this shouldn't be a problem?

Reply:
@msgmaxim The front-end does not have an option to change dca interval params. It is our quoting endpoint that decides on the dca params. if the swap amount is under the configuration amount (~3000usd) we only use 1 chunk so it won't be an issue on our front-end or in our sdk if they use it like we recommend.

Reply:
@zoheb391  The issue here is that the protocol might change the parameters: if the chosen number of chunks would result in chunks that are smaller than some minimum, we reduce the number of chunks accordingly. This can still happen with our FE. Example: User requests a quote for 10BTC and gets a dca quote for 1000 chunks. But then they only send 0.1BTC. In this case, the protocol would adjust the number of chunks to something smaller than 1000.

Reply:
ah i see now. yeah we're using the channel for dca params. So that might be a problem for us. We will need to make that switch.

Reply:
i'm guessing interval remains unchanged though?

Reply:
> i'm guessing interval remains unchanged though?

Yes, no change to the interval but I would still recommend using the swap request details rather than the channel details for this.